service/src/main/java/greencity/service/UserServiceImpl.java

```diff
@@ -444,14 +439,12 @@ public UserGoalResponseDto updateUserGoalStatus(Long userId, Long goalId, String
             .orElseThrow(() -> new WrongIdException(ErrorMessage.USER_NOT_FOUND_BY_ID + userId));
         if (user.getUserGoals().stream().anyMatch(o -> o.getId().equals(goalId))) {
             userGoal = userGoalRepo.getOne(goalId);
-            if (userGoal.getStatus().equals(GoalStatus.DONE)) {
-                userGoal.setStatus(GoalStatus.ACTIVE);
-                userGoal.setDateCompleted(null);
-                userGoalRepo.save(userGoal);
-            } else if (userGoal.getStatus().equals(GoalStatus.ACTIVE)) {
+            if (userGoal.getStatus().equals(GoalStatus.ACTIVE)) {
```

Comment:
GoalStatus.ACTIVE.equals(userGoal.getStatus()) - better to use this approach, because either you can get nullpointer

Reply:
changed